src/core/border_router/routing_manager.cpp

```diff
@@ -1708,39 +1675,45 @@ void RoutingManager::RxRaTracker::HandleRouterTimer(void)
 
     for (Router &router : mRouters)
     {
-        if (!router.IsReachable())
+        if (!router.ShouldCheckReachability() || (router.mTimeout > now))
         {
             continue;
         }
 
-        // Skip NS probes if the router is this device. This prevents
-        // issues where the platform might not be able to receive and
-        // process the NA messages from the local device itself.
+        router.mNsProbeCount++;
 
-        if (router.mIsLocalDevice)
+        if (router.IsReachable())
         {
-            continue;
+            router.mTimeout = now + ((router.mNsProbeCount < Router::kMaxNsProbes) ? Router::kNsProbeRetryInterval
+                                                                                   : Router::kNsProbeTimeout);
```

Comment:
Is `kNsProbeTimeout` actually an interval as well? Asking because if not you're adding an interval to a timestamp in one case and a timestamp to a timestamp in the other.

Reply:
Yes. It is a constant indicating timeout interval to use on last NS probe.